### PR TITLE
[server] Add tests for resource-access and env-var-service

### PR DESCRIPTION
**Pull request by geropl**
## Description

Improves tests around EnvVars

## Related Issue(s)

Related: WEB-353

#### Preview status

gitpod:summary

## Build Options

<details>
<summary>Build</summary>

- [ ] /werft with-werft
      Run the build with werft instead of GHA
- [ ] leeway-no-cache
- [ ] /werft no-test
      Run Leeway with `--dont-test`
</details>

<details>
<summary>Publish</summary>

- [ ] /werft publish-to-npm
- [ ] /werft publish-to-jb-marketplace
</details>

<details>
<summary>Installer</summary>

- [ ] analytics=segment
- [ ] with-dedicated-emulation
- [x] with-ws-manager-mk2
- [ ] workspace-feature-flags
  Add desired feature flags to the end of the line above, space separated
</details>

<details>
<summary>Preview Environment</summary>

- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [ ] /werft with-preview
- [ ] /werft with-large-vm
- [ ] /werft with-gce-vm
      If enabled this will create the environment on GCE infra
- [ ] with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`
</details>
 
/hold

### CODE REVIEW

File: components/server/src/auth/resource-access.spec.ts [hunk 1]
```diff
@@ -778,13 +778,6 @@ class TestResourceAccess {
                 operation: "get",
                 expectation: false,
             },
-            {
-                name: "explicit scope with exact same owner and repo",
-                guard: new WorkspaceEnvVarAccessGuard([getEnvVarResourceScope]),
```
Comment (geropl) [inline components/server/src/auth/resource-access.spec.ts:783]: Exact duplicate from two cases above

**Comment by geropl** [comment]
Re-running build which was failing due to an unrelated fluke.

**Comment by geropl** [comment]
oom'ed... :upside_down_face:

**Review by easyCZ — APPROVED** [review]

**Pull request merged**